libvips/foreign/pngsave.c

```diff
@@ -133,13 +136,37 @@ vips_foreign_save_png_class_init( VipsForeignSavePngClass *class )
 		VIPS_TYPE_FOREIGN_PNG_FILTER,
 		VIPS_FOREIGN_PNG_FILTER_ALL );
 
+	VIPS_ARG_INT( class, "colors", 13,
+		_( "Colors" ),
+		_( "Number of palette entries" ),
+		VIPS_ARGUMENT_OPTIONAL_INPUT,
+		G_STRUCT_OFFSET( VipsForeignSavePng, colors ),
+		2, 256, 256 );
```

Comment:
Should a value of `0`, meaning don't quantise, be permitted here?

Reply:
The range for `max_colors` in libimagequant is 2-256 so we can't reasonably permit `0` here, because it would also allow `1` (at least if we want to use the default option range checks).

Internally though `colors` is initialized to `0` which means don't quantize.

My thought was that people should simply omit the `colors` option if quantization is unwanted.

Reply:
Oh well, we could get away with redefining the range to 0-255 and simply doing a `colors + 1` when the value is passed to libimagequant. That would be a bit weird for users though, cause they shouldn't be counting from zero.

Reply:
The 2-256 range is fine as-is. As you say, anything else would be too confusing.

Reply:
Not too fond of the name `quant`, how about `palette`?

Reply:
Yes, that sounds better.

Reply:
Quick thought: should the `colors` property be made consistent with the British English "colour" spelling used everywhere else in the libvips API e.g. `colourspace`? (The approach taken by sharp is to provide aliases so both work.)

Reply:
Good point, libvips is en_GB for API names. I noticed this too, then forgot to comment :( More coffee.